### PR TITLE
RoomOverviewScreen

### LINKED ISSUE #183: room overview screen
A List of all rooms with their beds and assigned tasks.

Wait for the design of this screen before implementing
room-overview

### PR DESCRIPTION
**Pull request by use-to**
Closes #183 

Strongly depends on #191 as there are so many undefined colors and widget sizes.

### CODE REVIEW

File: apps/tasks/lib/screens/room_overview_screen.dart [hunk 1]
```diff
@@ -1,47 +1,243 @@
...
+                    .map((PatientDTO patient) => BedCard(patient: patient))
+                    .toList(),
+              )
+            : Padding(
+                padding: const EdgeInsets.only(left: 20, top: 10, bottom: 10),
+                child: Text(
+                  context.localization!.noBedAvailable,
+                  style: const TextStyle(
+                    fontWeight: FontWeight.w300,
+                  ),
```
Comment (DasProffi) [inline apps/tasks/lib/screens/room_overview_screen.dart:83]: dont use custom fontWeight use a TextStyle from texttheme in stead or mark with a todo for the texttheme/themeUpdate

**Review by DasProffi — CHANGES_REQUESTED** [review]

**Comment by DasProffi** [comment]
The Screen and BedCards look great

**Comment by florian-sabonchi** [comment]
What is your perspective on ascribing the Pull Request a provisional classification of 'Draft' in instances when it is not presently undergoing proactive modifications?

File: apps/tasks/lib/main.dart [hunk 2]
```diff
@@ -39,7 +40,9 @@ class MyApp extends StatelessWidget {
           ],
           supportedLocales: getSupportedLocals(),
           locale: Locale(languageNotifier.language),
-          home: const LandingScreen(),
+          home: const SafeArea(
+            child: LandingScreen(),
+          ),
```
[the remark this answers is not in the document]
Comment (use-to) [reply apps/tasks/lib/main.dart:45]: Applying "ignored for now" cause bn-bar should be fine.

File: apps/tasks/lib/components/task_status_pill.dart [hunk 3]
```diff
@@ -57,11 +57,17 @@ class TaskStatusPill extends StatelessWidget {
...
+              color: dotColor ?? Colors.black,
+              diameter: 7.5,
+            ),
+            const SizedBox(width: 4),
             Text(
               text,
-              style: TextStyle(color: textColor ?? Colors.black),
+              style: TextStyle(
+                color: textColor ?? Colors.black,
+                fontSize: 12.5,
```
[the remark this answers is not in the document]
Comment (use-to) [reply apps/tasks/lib/components/task_status_pill.dart:69]: Also marked as `// TODO` for theming update

File: apps/tasks/lib/screens/room_overview_screen.dart [hunk 4]
```diff
@@ -1,47 +1,249 @@
...
+            Icons.filter_alt,
+            color: Colors.white,
           ),
+          onPressed: () => print("PRESSED"), // TODO open filter modal
         ),
-      ),
-    );
+        body: ListView(
+          shrinkWrap: true,
+          padding: const EdgeInsets.only(bottom: 25),
```
Comment (DasProffi) [inline apps/tasks/lib/screens/room_overview_screen.dart:243]:
```suggestion
          padding: const EdgeInsets.only(bottom: distanceDefault),
```

File: apps/tasks/lib/components/task_status_pill_box.dart [hunk 5]
```diff
@@ -53,6 +54,9 @@ class TaskStatusPillBox extends StatelessWidget {
         topRight: Radius.circular(borderRadius),
       ),
     );
+
+    const double paddingHorizontal = 6;
```
Comment (DasProffi) [inline apps/tasks/lib/components/task_status_pill_box.dart:58]:
```suggestion
    const double paddingHorizontal = paddingTiny;
```

add `const double paddingTiny= 6;` to constants file
or make the paddingHorizontal a attribute of the widget

File: apps/tasks/lib/components/task_status_pill.dart [hunk 6]
```diff
@@ -57,11 +57,18 @@ class TaskStatusPill extends StatelessWidget {
             shape: shape ?? const RoundedRectangleBorder(),
           ),
           child: Row(children: [
-            Circle(color: dotColor ?? Colors.black),
-            const SizedBox(width: distanceTiny),
+            Circle(
+              color: dotColor ?? Colors.black,
+              diameter: 7.5,
+            ),
+            const SizedBox(width: 4),
```
Comment (DasProffi) [inline apps/tasks/lib/components/task_status_pill.dart:64]: Make this an attribute of the widget or use a predefined style constant like `distanceTiny`

**Review by DasProffi — CHANGES_REQUESTED** [review]

**Review by DasProffi — APPROVED** [review]

**Pull request merged**